### DISCUSSION TITLE
An issue about CT scan data augmentation

### DISCUSSION THREAD

**Discussion by lyangfan** [TorchIO-project/torchio#668] [Q&A]
Hi all,
      I'm a new comer of medical images field. I want to finish a segmentation task of CT scans. My dataset size is limited so I want to augment it. I'm confused which augmentation method I should use because I learn that torchIO is slightly biased torwards MRI at YouTube. So can you give some suggestions about the augmentation way or how to find the best augmentation method?
P.S: The dataset is whole body scan(not human, but pigs). I have two kinds of objects which are needed to be segmented. One is cavity(thoracic cavity, abdomen cavity, pelvic cavity), the other one is specific muscle tissue(like musculus biceps brachii). As you can see, one is bigger anatomically and the other one is smaller.
       Looking forward to your reply.
Yangfan

**Comment by fepegar** [comment]
Hi, @lyangfan. I suggest you start training without augmentation and move on to augment your data once you start overfitting.

CT images are typically large because of their high resolution and field of view, so you could look into [Patch-based pipelines](https://torchio.readthedocs.io/data/patch_based.html). If you want to focus the patches sampling on your structures of interest, you could use a [`WeightedSampler`](https://torchio.readthedocs.io/data/patch_training.html#torchio.data.WeightedSampler).

Maybe flipping doesn't make sense because of the asymmetry across all the dimensions of your dataset. You can start with some generic transforms, for example:

```python
import torchio as tio
preprocess = tio.RescaleIntensity((-1, 1))
augmentation = tio.Compose([
    tio.RandomAffine(),
    preprocess,
    tio.RandomNoise(),
])
```

**Reply by lyangfan** [reply]
Thanks for your reply. I will try.

**Reply by fepegar** [reply]
No worries!
